### PR TITLE
Complete Rework

### PR DESCRIPTION
**Pull request by flhps**
This should be a largely functional version that could be the basis for the first open-source release. First step is testing and completing this PR.

### CODE REVIEW

**Comment by jdsika** [comment]
@jfelixh be aware of [this branch](https://github.com/GAIA-X4PLC-AAD/gx-credentials/tree/license_and_Material_UI).

Add the license and the SPDX headers in the rework branch and delete this outdated one.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "prettier.prettierPath": "./node_modules/prettier"
```
Comment (jdsika) [inline .vscode/settings.json]: vscode settings should not be included. "prettier" should be part of the GitHub "workflow"

**Comment by flhps** [comment]
Will do tomorrow and then it should be ready to open-source from my perspective.

File: cache/config.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+	"telemetry": {
+		"notifiedAt": "1682077309825",
+		"enabled": false
```
Comment (jdsika) [inline cache/config.json:4]: Where is the cache coming from? Should this be part of the gitignore?

**Review by jdsika — CHANGES_REQUESTED** [review]

File: cache/config.json (hunk 2 above)
Comment (flhps) [reply cache/config.json:4]: This is the result of turning telemetry reporting off for Next.js. You can read on it here: https://nextjs.org/telemetry
I think it should be committed if we all agree on turning telemetry off for the project. It is somewhat pointless to only turn it off on one or two developer machines if a third one reports project metadata.

File: .vscode/settings.json (hunk 1 above)
Comment (flhps) [reply .vscode/settings.json]: Removed settings from tracking and added github action for prettier formatting.

**Comment by flhps** [comment]
Everything should be ready now. Feel free to do further adjustments or merge, @jdsika

File: .vscode/settings.json (hunk 1 above)
Comment (flhps) [reply .vscode/settings.json]: We might want to add "development" to also receive prettier formatting. Right now it only is set to affect "main".

**Review by jdsika — APPROVED** [review]

**Comment by jdsika** [comment]
Good job everyone!

**Pull request merged**